docs/tutorial/snapcraft-guide.md

```diff
@@ -0,0 +1,181 @@
+distribution.
+
+# Using `electron-installer-snap`
+
+The module works like `electron-winstaller` and similar modules in that its
+scope is limited to building snap packages. You can install it with:
+
+```sh
+npm install --save-dev electron-installer-snap
+```
+
+## Step 1: Package Your Electron Application
+
+Package the application using [electron-packager][electron-packager] (or a
+similar tool). Make sure to remove `node_modules` that you don't need in your
+final application, since any module you don't actually need will just increase
+your application's size.
+
+The output should look roughly like this:
+
+```text
+├── Ghost.exe
+├── LICENSE
+├── content_resources_200_percent.pak
+├── content_shell.pak
+├── d3dcompiler_47.dll
+├── ffmpeg.dll
+├── icudtl.dat
+├── libEGL.dll
+├── libGLESv2.dll
+├── locales
+│   ├── am.pak
+│   ├── ar.pak
+│   ├── [...]
+├── natives_blob.bin
+├── node.dll
+├── resources
+│   ├── app
+│   └── atom.asar
+├── snapshot_blob.bin
+├── squirrel.exe
+└── ui_resources_200_percent.pak
+```
+
+## Step 2: Running electron-installer-snap
+
+From a terminal that has `snapcraft` in its `PATH`, run `electron-installer-snap`
+with the only required parameter `--out`, which is the location of your packaged
+Electron application created in the first step.
+
+```sh
+npx electron-installer-snap --src=out/myappname-linux-x64
+```
+
+If you have an existing build pipeline, you can use `electron-installer-snap`
+programmatically. For more information, see the API docs.
+
+```js
+const snap = require('electron-installer-snap')
+
+snap(options)
+  .then(snapPath => console.log(`Created snap at ${snapPath}!`))
+```
+
+# Using an Existing Debian Package
+
+Snapcraft is capable of taking an existing `.deb` file and turning it into
+a `.snap` file. The creation of a snap is configured using a `snapcraft.yaml`
+file that describes the sources, dependencies, description, and other core
+building blocks.
+
+## Step 1: Create a Debian Package
+
+If you do not already have a `.deb` package, using `electron-installer-snap`
+might be an easier path to create snap packages. However, multiple solutions
+for creating Debian packages exist, including [`electron-forge`](electron-forge),
+[`electron-builder`]() or [`electron-installer-debian`](electron-installer-debian).
+
+## Step 2: Create a snapcraft.yaml
+
+For more information on the available configuration options, see the
+[documentation on the snapcraft syntax](https://docs.snapcraft.io/build-snaps/syntax).
+In this example
+
+```yaml
+name: myApp
+version: 2.0.0
+summary: A little description for the app.
+description: |
+ You know what? This app is amazing! It does all the things
+ for you. Some say it keeps you young, maybe even happy.
+
+grade: stable
+confinement: classic
+
+parts:
+  slack:
+    plugin: dump
+    source: my-deb.deb
+    source-type: deb
+    after:
+      - desktop-gtk2
+    build-packages:
+      - patchelf
```

Comment:
is there are reason for adding patchelf here?
@sergiusens has worked to make it a snapcraft dependency that just works everywhere, so this shouldn't be needed.

Reply:
Currently needed to build Slack – can you 100% confirm that it's not needed?

Reply:
I can confirm 100% that it **shouldn't** be needed. We'll have to test if 2.39 actually fixes it.

Reply:
FWIW that has never been part of `electron-installer-snap`.

Reply:
Alright, taking it out.